### PR TITLE
[example] add llama2 example

### PR DESCRIPTION
**Pull request by ver217**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: examples/language/llama2/scripts/benchmark_7B/hosts.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+192.168.0.38
+192.168.0.39
+192.168.0.40
+192.168.0.41
```
Comment (FrankLeeeee) [inline examples/language/llama2/scripts/benchmark_7B/hosts.txt:4]: Remove this file?

File: examples/language/llama2/scripts/benchmark_7B/gemini_auto.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+cd ../..
+
+# NCCL IB environment variables
+export NCCL_IB_HCA=mlx5_1:1,mlx5_2:1,mlx5_3:1,mlx5_4:1
+export NCCL_IB_DISABLE=0
+export NCCL_SOCKET_IFNAME=eth0
+export NCCL_IB_GID_INDEX=3
+export NCCL_IB_TIMEOUT=23
+export NCCL_IB_RETRY_CNT=7
```
Comment (FrankLeeeee) [inline examples/language/llama2/scripts/benchmark_7B/gemini_auto.sh:17]: Should not set these variables by default as they vary on different machines

File: examples/language/llama2/scripts/benchmark_7B/gemini.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+cd ../..
+
+# NCCL IB environment variables
+export NCCL_IB_HCA=mlx5_1:1,mlx5_2:1,mlx5_3:1,mlx5_4:1
+export NCCL_IB_DISABLE=0
+export NCCL_SOCKET_IFNAME=eth0
+export NCCL_IB_GID_INDEX=3
+export NCCL_IB_TIMEOUT=23
+export NCCL_IB_RETRY_CNT=7
```
Comment (FrankLeeeee) [inline examples/language/llama2/scripts/benchmark_7B/gemini.sh:17]: Same.

File: examples/language/llama2/scripts/benchmark_7B/hosts.txt (hunk 1 above)
Comment (ver217) [reply examples/language/llama2/scripts/benchmark_7B/hosts.txt:4]: BUT we still need a sample. Should I add this to README?

Comment (FrankLeeeee) [reply examples/language/llama2/scripts/benchmark_7B/hosts.txt:4]: Yes, should provide an example in README.

Comment (FrankLeeeee) [reply examples/language/llama2/scripts/benchmark_7B/hosts.txt:4]: The NCCL variables can go to README as well.

**Review by binmakeswell — APPROVED** [review]

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**